icicle-compiler/src/Icicle/Sea/FromAvalanche/Base.hs

```diff
@@ -35,25 +40,75 @@ import           Text.Printf (printf)
 
 ------------------------------------------------------------------------
 
-textOfName :: Pretty n => n -> Text
-textOfName = T.pack . show . seaOfName
-
-seaOfName :: Pretty n => n -> Doc
-seaOfName = string . fmap mangle . show . pretty
+-- | A legal C identifier.
+newtype SeaName = SeaName {
+    getSeaName :: Text
+ } deriving (Eq, Ord, Show)
+
+-- | A value that will need be quoted in C.
+newtype SeaString = SeaString {
+    getSeaString :: Text
+ } deriving (Eq, Ord, Show)
+
+instance Pretty SeaName where
+  pretty =
+    string . Text.unpack . takeSeaName
+
+instance Pretty SeaString where
+  pretty =
+    string . Text.unpack . getSeaString
```

Comment:
hm, maybe it shouldn't have a pretty instance then - if it has to be explicitly handled?